### PR TITLE
Normalize with intrinsics

### PR DESCRIPTION
**Pull request by ducha-aiki**
#### Changes
I have refactored out part of project/unproject points to `normalize_with_intrinsics`. 
The latter function (in their own implementation) is heavily used on OANet and other correspondences filtering  methods, e.g.

https://github.com/zjhthu/OANet/blob/51d71ff3f57161e912ec72420cd91cf7db64ab74/demo/learnedmatcher.py#L72
also is used in IMC eval

https://github.com/ubc-vision/image-matching-benchmark/blob/master/utils/stereo_helper.py#L54

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by edgarriba — COMMENTED** [review]
I would keep it under camera since it's specific implementation of pinhole models -- our maybe we could take this as opportunity to create a dedicated camera.pinhole.py. Specially, since i Ihave some plans to add more camera representations. Besides, conversions.py I believe should be refactored at some point, also because all the geometry conversions there probably will be replaced by the geometry.liegroup im planning with Sophus author.

**Comment by ducha-aiki** [comment]
Maybe in future, but figuring out now where to put it, would result in many cyclic dependencies between perspective, pinhole, etc camera. 
I am also not a fan of using classes everywhere, as for Pinhole camera, when in the many cases user just need a single simple function (as you can see in the IMC code example)

**Comment by ducha-aiki** [comment]
Also, I am against putting conversions into liegroups without keeping some alias from the more general module. 
Reason - in many cases user have no idea wtf is the Lie group and therefore would never look for simple intrinsic-normalization function there

**Comment by ducha-aiki** [comment]
@edgarriba what is happening with mypy test of `augmentation.container`?

image

```
=================================== FAILURES ===================================
[271](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:272)
_________________________________ test session _________________________________
[272](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:273)
mypy exited with status 1.
[273](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:274)
___________________ kornia/augmentation/container/augment.py ___________________
[274](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:275)
247: error: Invalid index type
[275](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:276)
____________________ kornia/augmentation/container/image.py ____________________
[276](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:277)
347: error: Invalid index type
[277](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:278)
356: error: Item "None" of
[278](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:279)
356: error: Argument 2 to "inverse" of
[279](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:280)
===================================== mypy =====================================
[280](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:281)
"Union[str, Any]" for "List[ParamItem]"; expected type "SupportsIndex"  [index]
[281](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:282)
                    param = params[name] if name in params else param
[282](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:283)
                                   ^
[283](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:284)
"Union[ParamItem, Any, None]" has no attribute "data"  [union-attr]
[284](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:285)
                    input = module.inverse(input, param.data, extra_args=e...
[285](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:286)
                                                  ^
[286](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:287)
"ImageSequential" has incompatible type
[287](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:288)
"Union[Dict[Any, Any], List[Any], None, Any]"; expected
[288](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:289)
"Optional[List[ParamItem]]"  [arg-type]
[289](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:290)
                    input = module.inverse(input, param.data, extra_args=e...
[290](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:291)
                                                  ^
[291](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:292)
"Union[str, Any]" for "List[ParamItem]"; expected type "SupportsIndex"  [index]
[292](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:293)
                        param = params[name] if name in params else param
[293](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:294)
                                       ^
[294](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:295)
Found 4 errors in 2 files (checked 256 source files)
[295](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:296)
=========================== short test summary info ============================
[296](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:297)
FAILED kornia/__init__.py::mypy-status
[297](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:298)
FAILED kornia/augmentation/container/augment.py::mypy
[298](https://github.com/kornia/kornia/runs/6656973962?check_suite_focus=true#step:4:299)
FAILED kornia/augmentation/container/image.py::mypy
```

**Comment by edgarriba** [comment]
@ducha-aiki something went wrong recently. I patched here: https://github.com/kornia/kornia/pull/1724

@shijianjian will check tomorrow

**Review by edgarriba — COMMENTED** [review]
LGTM. Just naming concern to have better code readability

File: kornia/geometry/conversions.py [hunk 1]
```diff
@@ -55,7 +57,7 @@ class QuaternionCoeffOrder(enum.Enum):
     WXYZ = 'wxyz'
 
 
-def rad2deg(tensor: torch.Tensor) -> torch.Tensor:
+def rad2deg(tensor: Tensor) -> Tensor:
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:60]: should we deprecate this one and below? https://pytorch.org/docs/stable/generated/torch.deg2rad.html

File: kornia/geometry/conversions.py [hunk 2]
```diff
@@ -111,15 +113,15 @@ def pol2cart(rho: torch.Tensor, phi: torch.Tensor) -> Tuple[torch.Tensor, torch.
         >>> phi = torch.rand(1, 3, 3)
         >>> x, y = pol2cart(rho, phi)
     """
-    if not (isinstance(rho, torch.Tensor) & isinstance(phi, torch.Tensor)):
-        raise TypeError(f"Input type is not a torch.Tensor. Got {type(rho)}, {type(phi)}")
+    if not (isinstance(rho, Tensor) & isinstance(phi, Tensor)):
+        raise TypeError(f"Input type is not a Tensor. Got {type(rho)}, {type(phi)}")
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:117]: KORNIA_CHECK ?

File: kornia/geometry/conversions.py [hunk 3]
```diff
@@ -879,31 +871,27 @@ def normalize_pixel_coordinates(
...
+        >>> coords = tensor([[50., 100.]])
         >>> normalize_pixel_coordinates(coords, 100, 50)
         tensor([[1.0408, 1.0202]])
     """
     if pixel_coordinates.shape[-1] != 2:
         raise ValueError("Input pixel_coordinates must be of shape (*, 2). " "Got {}".format(pixel_coordinates.shape))
 
     # compute normalization factor
-    hw: torch.Tensor = torch.stack(
+    hw: Tensor = stack(
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:882]: possibly all this group of functions to do something like ?
```python
hw = tensor([width, height], device=pixel_coordinates.device, dtype=pixel_coordinates.dtype)
```

Comment (edgarriba) [reply kornia/geometry/conversions.py:882]: to avoid stack and allocate directly from device/dtype?

File: kornia/geometry/conversions.py [hunk 4]
```diff
@@ -1012,14 +992,14 @@ def angle_to_rotation_matrix(angle: torch.Tensor) -> torch.Tensor:
         >>> output = angle_to_rotation_matrix(input)  # Nx3x2x2
     """
     ang_rad = deg2rad(angle)
-    cos_a: torch.Tensor = torch.cos(ang_rad)
-    sin_a: torch.Tensor = torch.sin(ang_rad)
-    return torch.stack([cos_a, sin_a, -sin_a, cos_a], dim=-1).view(*angle.shape, 2, 2)
+    cos_a: Tensor = torch.cos(ang_rad)
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:995]:
```suggestion
    cos_a: Tensor = ang_rad.cos()
```

File: kornia/geometry/conversions.py [hunk 5]
```diff
@@ -1012,14 +992,14 @@ def angle_to_rotation_matrix(angle: torch.Tensor) -> torch.Tensor:
...
+    sin_a: Tensor = torch.sin(ang_rad)
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:996]:
```suggestion
    sin_a: Tensor = ang_rad.sin()
```

File: kornia/geometry/conversions.py [hunk 6]
```diff
@@ -1179,15 +1159,100 @@ def normalize_homography3d(
...
-    dst_norm_trans_dst_pix: torch.Tensor = normal_transform_pixel3d(dst_d, dst_h, dst_w).to(dst_pix_trans_src_pix)
+    dst_norm_trans_dst_pix: Tensor = normal_transform_pixel3d(dst_d, dst_h, dst_w).to(dst_pix_trans_src_pix)
     # compute chain transformations
-    dst_norm_trans_src_norm: torch.Tensor = dst_norm_trans_dst_pix @ (dst_pix_trans_src_pix @ src_pix_trans_src_norm)
+    dst_norm_trans_src_norm: Tensor = dst_norm_trans_dst_pix @ (dst_pix_trans_src_pix @ src_pix_trans_src_norm)
     return dst_norm_trans_src_norm
 
 
+def normalize_points_with_intrinsics(point_2d: Tensor, camera_matrix: Tensor):
+    r"""Normalizes points with intrinsics. Useful for conversion of keypoints to be used with essential matrix.
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:1172]:
```suggestion
    r"""Normalizes points with intrinsics.
    
    Useful for conversion of keypoints to be used with essential matrix.
```

File: kornia/geometry/conversions.py [hunk 7]
```diff
@@ -1179,15 +1159,100 @@ def normalize_homography3d(
...
+    # projective
+    x_coord: Tensor = (u_coord - cx) / fx
+    y_coord: Tensor = (v_coord - cy) / fy
+
+    xy: Tensor = stack([x_coord, y_coord], dim=-1)
+    return xy
+
+
+def denormalize_points_with_intrinsics(point_2d_norm: Tensor, camera_matrix: Tensor):
+    r"""Normalizes points with intrinsics. Useful for conversion of keypoints to be used with essential matrix.
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:1215]:
```suggestion
    r"""Normalizes points with intrinsics.
    
    Useful for conversion of keypoints to be used with essential matrix.
```

File: kornia/geometry/quaternion.py [hunk 8]
```diff
@@ -325,9 +325,6 @@ def random(cls, batch_size: int) -> 'Quaternion':
 
         Example:
             >>> q = Quaternion.random(batch_size=2)
-            >>> q.norm()
```
Comment (edgarriba) [inline kornia/geometry/quaternion.py:328]: ??

File: kornia/testing/__init__.py [hunk 9]
```diff
@@ -279,7 +279,7 @@ def KORNIA_CHECK_SHAPE(x, shape: List[str]) -> None:
         start_idx: int = 1
```
Comment (edgarriba) [inline kornia/testing/__init__.py:279]:
```suggestion
        start_idx = 1
```
define the variable before the if branch as `start_idx: int`

File: test/geometry/test_conversions.py [hunk 10]
```diff
@@ -1381,6 +1381,153 @@ def test_jit(self, device, dtype):
         assert_close(actual, expected)
 
 
+class TestProjectPoints:
```
Comment (edgarriba) [inline test/geometry/test_conversions.py:1384]:
```suggestion
class TestProjectPoints(BaseTester):
```

then you can use `self.assert_close` with handles some dtype tolerance amongst others

File: kornia/geometry/conversions.py (hunk 1 above)
Comment (ducha-aiki) [reply kornia/geometry/conversions.py:60]: Not deprecate, but just pass-through to the PyTorch function. I don't like breaking compatibility
